### PR TITLE
deck helmets

### PR DESCRIPTION
**Pull request by Greenjoe12345**
Adds deck helmets, which can be found in the loadout, and can be found in hanger techs lockers.
These have very minor armor
(also adds hazard vests to hanger tech lockers)
image

### Asset Licenses
The following assets that **have not** been created by myself are included in this PR:

| Path | Original Author | License |
| icons/clothing/head/deckhelmet.dmi | JoeyNosegay - Baystation12 | AGPL 3.0 |

### CODE REVIEW

**Comment by Greenjoe12345** [comment]
!review

**Comment by WickedCybs** [comment]
I think giving it the same resistances as the first responder's helmet would be fine. They can get those in the loadout I think and It's really nothing major. They're both plastic helmets.

File: code/modules/clothing/head/deck_helmets.dm [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	flags_inv = BLOCKHEADHAIR
+	contained_sprite = TRUE
+	armor = list(
+		melee = ARMOR_MELEE_SMALL,
+		bullet = ARMOR_BALLISTIC_MINOR,
+		laser = ARMOR_LASER_MINOR,
+		energy = ARMOR_ENERGY_MINOR,
+		bomb = ARMOR_BOMB_PADDED,
+		bio = ARMOR_BIO_MINOR,
+		rad = ARMOR_RAD_MINOR
```
Comment (KingOfThePing) [inline code/modules/clothing/head/deck_helmets.dm:16]: I dont think these -plastic- helmets should have any armor values, except ARMOR_MELEE_SMALL, if at all.

**Review by KingOfThePing — CHANGES_REQUESTED** [review]

File: code/modules/clothing/head/deck_helmets.dm (hunk 1 above)
Comment (Greenjoe12345) [reply code/modules/clothing/head/deck_helmets.dm]: fair point tbh, was thinking that myself

**Comment by KingOfThePing** [comment]
I actually didnt knew the FR helmets have any resistances. They shouldn't either, imo. In the end its probably minor and only my personal opinion.

**Review by NonQueueingMatt — APPROVED** [review]

**Comment by DreamySkrell** [comment]
Does every hangar tech locker need 5 helmets each? That is 15 helmets, that to be honest, I barely imagine even one would be used in a round with hangar techs.

One helmet in each locker should be enough imo. Anyone else can get it from the loadout.

File: code/game/objects/structures/crates_lockers/closets/secure/operations.dm [hunk 2]
```diff
@@ -47,6 +49,12 @@
 	new /obj/item/modular_computer/handheld/preset/supply/cargo_delivery(src)
 	new /obj/item/export_scanner(src)
 	new /obj/item/device/flashlight/marshallingwand(src)
+	new /obj/item/clothing/head/deckhelmet(src)
+	new /obj/item/clothing/head/deckhelmet/green(src)
```
Comment (alsoandanswer) [inline code/game/objects/structures/crates_lockers/closets/secure/operations.dm:53]: Get rid of everything else other than the default helmet if you're going to put them in the loadout anyway.

File: code/modules/client/preference_setup/loadout/loadout_head.dm [hunk 3]
```diff
@@ -375,6 +375,21 @@
 	path = /obj/item/clothing/head/hairnet
 	flags = GEAR_HAS_NAME_SELECTION | GEAR_HAS_DESC_SELECTION | GEAR_HAS_COLOR_SELECTION
 
+/datum/gear/head/deckhelmet
+	display_name = "deck helmet selection"
+	path = /obj/item/clothing/head/deckhelmet
+	flags = GEAR_HAS_NAME_SELECTION | GEAR_HAS_DESC_SELECTION
```
Comment (alsoandanswer) [inline code/modules/client/preference_setup/loadout/loadout_head.dm:381]: I think you should do away with all the variations and use GEAR_HAS_COLOR_SELECTION instead.

If you are insistent on using the current system, then please use the already estabilished color palette from the existing departmental uniforms.

**Review by alsoandanswer — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/crates_lockers/closets/secure/operations.dm (hunk 2 above)
Comment (Greenjoe12345) [reply code/game/objects/structures/crates_lockers/closets/secure/operations.dm:53]: That's fair enough, I'd just been thinking of a way to have the other kinds available outside of the loadout, too

File: code/modules/client/preference_setup/loadout/loadout_head.dm (hunk 3 above)
Comment (Greenjoe12345) [reply code/modules/client/preference_setup/loadout/loadout_head.dm:381]: They're meant to be made of brightly colored plastic, but what sort of color hexes should I use for them?

**Comment by Greenjoe12345** [comment]
could I get replies on the stuff I said in the requested changes?

**Comment by WickedCybs** [comment]
> could I get replies on the stuff I said in the requested changes?

Feel free to ping anyone who requests changes that you've done it/responded if there's been no check in for a while. Use the coding channel for that.

**Comment by Greenjoe12345** [comment]
> 5 (or 6) helmets is a bit overkill, even in the loadout. consider just one helmet that spawns in the OM and deck technicians' lockers, with the same colour as the operations uniforms.

I'll strip it down to one in each locker I guess then
I was considering sticking some of the colored ones around on the map itself in a future PR, such as the tug operator's one by the cargo tug and the munitions one by the munitions storage, once we get that...

File: code/modules/client/preference_setup/loadout/loadout_head.dm [hunk 4]
```diff
@@ -375,6 +375,21 @@
...
+
+/datum/gear/head/deckhelmet/New()
+	..()
+	var/list/deckhelmet = list()
+	deckhelmet["deck helmet"] = /obj/item/clothing/head/deckhelmet
+	deckhelmet["green deck helmet"] = /obj/item/clothing/head/deckhelmet/green
+	deckhelmet["blue deck helmet"] = /obj/item/clothing/head/deckhelmet/blue
+	deckhelmet["yellow deck helmet"] = /obj/item/clothing/head/deckhelmet/yellow
+	deckhelmet["purple deck helmet"] = /obj/item/clothing/head/deckhelmet/purple
+	//deckhelmet["red deck helmet"] = /obj/item/clothing/head/deckhelmet/red //uncomment when we get shipguns
```
Comment (SleepyGemmy) [inline code/modules/client/preference_setup/loadout/loadout_head.dm:391]: don't leave in uncommented code. either uncomment it or remove it.

File: code/modules/clothing/head/deck_helmets.dm [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	icon_state = "deckcrew_y"
+	item_state = "deckcrew_y"
+
+/obj/item/clothing/head/deckhelmet/purple
+	name = "purple deck helmet"
+	desc = "A plastic helmet with a visor, used by fueling personnel in hangars on many ships."
+	icon_state = "deckcrew_p"
+	item_state = "deckcrew_p"
+
+ //uncomment when we get shipguns
```
Comment (SleepyGemmy) [inline code/modules/clothing/head/deck_helmets.dm:37]: ditto.

**Comment by JohnWildkins** [comment]
Unaddressed reviews and merge conflict since 10/20, re-open when ready

**Pull request closed without merging**